sdk/provisioning/Azure.Provisioning/src/authorization/RoleAssignment.cs

```diff
@@ -44,16 +48,24 @@ public class RoleAssignment : Resource<RoleAssignmentData>
 
             AssignProperty(
                 data => data.Name,
-                $"guid('{resource.Name}', {(principalId == null ? "principalId" : "'" + principalId + "'")}," +
-                $" {SubscriptionResourceIdFunction}({(resource.Scope.Configuration?.UseInteractiveMode != true ? "'" + Id.SubscriptionId + "', ": string.Empty)}" +
-                $"'{RoleDefinitionResourceType}', '{roleDefinition}'))");
+                GetBicepName(resource));
 
             AssignProperty(
                 data => data.RoleDefinitionId,
                 $"{SubscriptionResourceIdFunction}({(resource.Scope.Configuration?.UseInteractiveMode != true ? "'"+ Id.SubscriptionId + "', ": string.Empty)}" +
                 $"'{RoleDefinitionResourceType}', '{roleDefinition}')");
         }
 
+        /// <inheritdoc />
+        protected override string GetBicepName(Resource resource)
+        {
+            var data = (RoleAssignmentData)ResourceData;
```

Comment:
nit:  why don't we use this.Properties here?

Reply:
because this is a Resource and not a `Resource<T>`